Tests/ApolloTests/Interceptors/MultipartResponseParsingInterceptorTests.swift

```diff
@@ -0,0 +1,355 @@
+      request: .mock(operation: MockSubscription.mock()),
+      response: .mock(
+        headerFields: ["Content-Type": "multipart/mixed;boundary=graphql"],
+        data: """
+          --graphql
+          content-type: test/custom
+
+          {
+            "data" : {
+              "key" : "value"
+            }
+          }
+          --graphql
+          """.crlfFormattedData()
+      )
+    ) { result in }
+
+    expect(requestChain.error as? MultipartResponseParsingInterceptor.MultipartResponseParsingError)
+      .to(equal(.unsupportedContentType(type: "test/custom")))
+  }
+
+  func test__error__givenChunk_withTransportError_shouldReturnError() throws {
+    let requestChain = ErrorRequestChain()
+
+    MultipartResponseParsingInterceptor().interceptAsync(
+      chain: requestChain,
+      request: .mock(operation: MockSubscription.mock()),
+      response: .mock(
+        headerFields: ["Content-Type": "multipart/mixed;boundary=graphql"],
+        data: """
+          --graphql
+          content-type: application/json
+
+          {
+            "errors" : [
+              {
+                "message" : "forced test failure!"
+              }
+            ],
+            "done": true
+          }
+          --graphql
+          """.crlfFormattedData()
+      )
+    ) { result in }
+
+    expect(requestChain.error as? MultipartResponseParsingInterceptor.MultipartResponseParsingError)
+      .to(equal(.irrecoverableError(message: "forced test failure!")))
+  }
+
+  func test__error__givenChunk_withMissingPayload_shouldReturnError() throws {
+    let requestChain = ErrorRequestChain()
+
+    MultipartResponseParsingInterceptor().interceptAsync(
+      chain: requestChain,
+      request: .mock(operation: MockSubscription.mock()),
+      response: .mock(
+        headerFields: ["Content-Type": "multipart/mixed;boundary=graphql"],
+        data: """
+          --graphql
+          content-type: application/json
+
+          {
+            "key": "value"
+          }
+          --graphql
+          """.crlfFormattedData()
+      )
+    ) { result in }
+
+    expect(requestChain.error as? MultipartResponseParsingInterceptor.MultipartResponseParsingError)
+      .to(equal(.cannotParsePayloadData))
+  }
+
+  func test__error__givenUnrecognizableChunk_shouldReturnError() throws {
+    let requestChain = ErrorRequestChain()
+
+    MultipartResponseParsingInterceptor().interceptAsync(
+      chain: requestChain,
+      request: .mock(operation: MockSubscription.mock()),
+      response: .mock(
+        headerFields: ["Content-Type": "multipart/mixed;boundary=graphql"],
+        data: """
+          --graphql
+          content-type: application/json
+
+          something
+          --graphql
+          """.crlfFormattedData()
+      )
+    ) { result in }
+
+    expect(requestChain.error as? MultipartResponseParsingInterceptor.MultipartResponseParsingError)
+      .to(equal(.cannotParseChunkData))
+  }
+
+  // MARK: Parsing tests
```

Comment:
The parsing tests involve more of the request chain mechanics to ensure that the interceptors all work together as expected.